src/coreclr/src/tools/crossgen2/ILCompiler.Reflection.ReadyToRun/PEReaderExtensions.cs

```diff
@@ -0,0 +1,126 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using System.Text;
+
+namespace ILCompiler.Reflection.ReadyToRun
+{
+    public class PEExportTable
+    {
+        private readonly Dictionary<string, int> _namedExportRva;
+        private readonly Dictionary<int, int> _ordinalRva;
+
+        private PEExportTable(PEReader peReader)
```

Comment:
How important is for this to be resilient against corrupted files?

Reply:
At this moment there are very few measures to guard against malformed files. According to my experience similar tools like PE / MSIL / PDB dumpers typically crash left and right when presented with malformed files; in my mental model this shouldn't pose a problem as they are typically local developer tools, not parts of the actual build pipeline, and end customer-facing app consistency is typically ensured by other means like signed envelopes.